pkg/storage/prunable/manager.go

```diff
@@ -8,41 +8,39 @@ import (
 	"github.com/iotaledger/hive.go/ds/shrinkingmap"
 	"github.com/iotaledger/hive.go/ierrors"
 	"github.com/iotaledger/hive.go/kvstore"
+	"github.com/iotaledger/hive.go/lo"
 	"github.com/iotaledger/hive.go/runtime/options"
 	"github.com/iotaledger/hive.go/runtime/syncutils"
 	"github.com/iotaledger/iota-core/pkg/model"
 	"github.com/iotaledger/iota-core/pkg/storage/database"
 	iotago "github.com/iotaledger/iota.go/v4"
 )
 
-type Manager struct {
-	openDBs      *cache.Cache[iotago.SlotIndex, *dbInstance]
-	openDBsMutex syncutils.Mutex
+type SlotManager struct {
+	openDBs      *cache.Cache[iotago.EpochIndex, *database.DBInstance]
+	openDBsMutex syncutils.RWMutex
 
-	lastPrunedSlot *model.EvictionIndex
-	pruningMutex   syncutils.RWMutex
+	lastPrunedEpoch *model.EvictionIndex[iotago.EpochIndex]
+	lastPrunedMutex syncutils.RWMutex
 
 	dbConfig     database.Config
 	errorHandler func(error)
 
-	dbSizes *shrinkingmap.ShrinkingMap[iotago.SlotIndex, int64]
+	dbSizes *shrinkingmap.ShrinkingMap[iotago.EpochIndex, int64]
 
-	// The granularity of the DB instances (i.e. how many buckets/slots are stored in one DB).
-	optsGranularity int64
-	optsMaxOpenDBs  int
+	optsMaxOpenDBs int
 }
 
-func NewManager(dbConfig database.Config, errorHandler func(error), opts ...options.Option[Manager]) *Manager {
-	return options.Apply(&Manager{
-		optsGranularity: 10,
-		optsMaxOpenDBs:  10,
+func NewSlotManager(dbConfig database.Config, errorHandler func(error), opts ...options.Option[SlotManager]) *SlotManager {
```

Comment:
Why is it called `SlotManager` if it deals with `EpochIndex`?

Reply:
I renamed it to `BucketManager`